jupyterhub/templates/hub/service.yaml

```
@@ -25,8 +25,17 @@ spec:
   selector:
     {{- include "jupyterhub.matchLabels" . | nindent 4 }}
   ports:
-    - port: 8081
+    - name: http
+      port: 8081
       targetPort: http
       {{- with .Values.hub.service.ports.nodePort }}
       nodePort: {{ . }}
       {{- end }}
+      # Support for Hub-managed services in k8s
+    {{- range $name, $config := .Values.hub.services }}
+    {{- if hasKey $config "k8s_managed_port" }}
+    - name: {{ $name | replace "_" "-"}}
+      port: {{ get $config "k8s_managed_port" }}
+      targetPort: {{ get $config "k8s_managed_port" }}
+    {{- end }}
+    {{- end }}
```

Same kind of changes suggested as in the other code suggestion.

```suggestion
    {{- /* Expose ports for additional services running in the hub pod /*}}
    {{- range $name, $config := .Values.hub.services }}
    {{- if hasKey $config "k8s_managed_port" }}
    - name: {{ $name | replace "_" "-" }}
      port: {{ $config.k8s_managed_port }}
      targetPort: {{ $config.k8s_managed_port }}
    {{- end }}
    {{- end }}
```

---

The name has the following criteria on it to be valid according to [the k8s api ref](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#serviceport-v1-core).

Name must either be an empty string or pass DNS_LABEL validation:
- must be no more than 63 characters long.
- must consist of lower case alphanumeric characters or '-'.
- must start and end with an alphanumeric character.